treescript/tests/test_mercurial.py

```diff
@@ -263,6 +322,55 @@ def info(msg, *args):
     assert is_slice_in_list(("TAG1", "TAG2"), called_args[1][0])
 
 
+@pytest.mark.asyncio
+async def test_do_tagging_no_tags(config, task, mocker):
+    """do_tagging is noop when check_tags returns an empty list."""
+
+    async def check_tags(*args):
+        return []
+
+    async def run_command(config, *arguments, repo_path=None):
+        assert False, "we shouldn't hit this."
+
+    mocker.patch.object(mercurial, "run_hg_command", new=run_command)
+    mocked_tag_info = mocker.patch.object(mercurial, "get_tag_info")
+    mocked_tag_info.return_value = {"revision": "deadbeef", "tags": ["TAG1", "TAG2"]}
+    mocked_source_repo = mocker.patch.object(mercurial, "get_source_repo")
+    mocked_source_repo.return_value = "https://hg.mozilla.org/treescript-test"
+    mocked_dontbuild = mocker.patch.object(mercurial, "get_dontbuild")
+    mocked_dontbuild.return_value = False
+    mocker.patch.object(mercurial, "check_tags", new=check_tags)
+    await mercurial.do_tagging(config, task, config["work_dir"])
+
+
+# log_outgoing {{{1
+@pytest.mark.parametrize("output", ("hg output!", None))
+@pytest.mark.asyncio
+async def test_log_outgoing(config, task, mocker, output):
+    called_args = []
+
+    async def run_command(config, *arguments, repo_path=None, **kwargs):
+        called_args.append([tuple([config]) + arguments, {"repo_path": repo_path}])
+        if output:
+            return output
+
+    mocker.patch.object(mercurial, "run_hg_command", new=run_command)
+    mocked_source_repo = mocker.patch.object(mercurial, "get_source_repo")
+    mocked_source_repo.return_value = "https://hg.mozilla.org/treescript-test"
+    await mercurial.log_outgoing(config, task, config["work_dir"])
+
+    assert len(called_args) == 1
+    assert "repo_path" in called_args[0][1]
+    assert is_slice_in_list(("out", "-vp"), called_args[0][0])
+    assert is_slice_in_list(("-r", "."), called_args[0][0])
+    assert is_slice_in_list(
+        ("https://hg.mozilla.org/treescript-test",), called_args[0][0]
+    )
```

Comment:
why not `assert "https://hg.mozilla.org/treescript-test" in called_args[0][0]` here since we don't need the is_slice_in_list if only one arg being tested...

Reply:
Done.